enhancements/ingress/optional-ingress.md

```diff
@@ -0,0 +1,247 @@
+---
+title: optional-ingress
+authors:
+  - "@alebedev87"
+reviewers:
+  - "@Miciah"
+  - "@bparees"
+  - "@cewong"
+  - "@wking"
+approvers:
+  - "@Miciah"
+  - "@bparees"
+  - "@wking"
+api-approvers:
+  - N/A
+creation-date: 2023-06-02
+last-updated: 2023-06-02
+tracking-link:
+  - https://issues.redhat.com/browse/NE-1129
+see-also:
+  - "/enhancements/installer/component-selection.md"
+  - "/enhancements/ingress/aws-load-balancer-operator.md"
+replaces:
+superseded-by:
+---
+
+# Make Ingress Operator optional
+
+## Summary
+
+This enhancement describes the implementation of a new capability to enable/disable the default ingress of the OpenShift cluster.
+The new capability (from here on called the ingress capability) includes [the cluster ingress operator](https://github.com/openshift/cluster-ingress-operator) as well as the custom resources managed by the operator.
+This enhancement focuses on the requirements for [RFE-3395](https://issues.redhat.com/browse/RFE-3395) which targets the ROSA clusters spawned and managed by HyperShift. The implementation of the ingress capability on the standalone OpenShift is out of the scope of this enhancement.
+
+## Motivation
+
+Users of the ROSA managed OpenShift service would like to use [AWS load balancers](/enhancements/ingress/aws-load-balancer-operator.md) as a way to accept the ingress traffic into the cluster.
+The cluster ingress operator becomes unnecessary and can be disabled to:
+- reduce the resource consumption on
+  - the HyperShift client clusters (ingress controllers)
+  - the HyperShift hosted control plane (operator)
+- make the HyperShift hosted control plane lighter, simpler therefore more manageable
+- enable the HyperShift cluster users to make their choice for the cluster ingress
+
+### Goals
+- Implement the new ingress capability.
+- Add the ingress capability to the ingress operator's payload.
+- Make the dependent cluster operators tolerate the absence of the ingress operator.
```

Comment:
I'd like to see a goal that ensures we intent to preserve a technical path for the same semantics introduced here to be reused in the future to express intent for disabling it in regular ocp even though we don't want to implement it today. I.e avoiding UX skew.

Also I'd like to see elaborated somewhere, maye be in the user stories why is this desired in ROSA but not in regular OCP, how are we justifying this feature skew between form factors? By reading the proposal as it is the motivation would be as beneficial for regular OCP as well, so it seems arbitrary.

Reply:
> I'd like to see a goal that ensures we intent to preserve a technical path for the same semantics introduced here to be reused in the future to express intent for disabling it in regular ocp even though we don't want to implement it today. I.e avoiding UX skew.

A goal is added.

> Also I'd expect to see elaborated somewhere, maye be in the user stories why is this desired in ROSA but not in regular OCP, how are we justifying this feature skew between form factors? By reading the proposal as it is the motivation would be as beneficial for regular OCP as well, so it seems arbitrary.

It's the customer request driven. Maybe renaming the EP to mention HyperShift can help to set up the reader's expectations well.